geowave/src/main/java/org/locationtech/sfcurve/geowave/geotime/index/dimension/TimeDefinition.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2013-2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License,
+ * Version 2.0 which accompanies this distribution and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0.txt
+ ******************************************************************************/
+package org.locationtech.sfcurve.geowave.geotime.index.dimension;
```

Comment:
I'd opine that naming any of our projects in the package names is inappropriate.  We haven't done it for GeoMesa or GeoTrellis.

Reply:
I thought about this, but also the GeoWave code is another implementation of Hilbert. What's unique about this approach is that it takes GeoWave's strategy specifically. I think it will make it a lot easier to reference and have everyone know what we are talking about. GeoTrellis's Z curve implementation was the original Z curve, and was only a Z curve - it didn't include a lot of the bells and whistles that the geowave-index project contains (tiering, periodicity). GeoMesa's bells and whistles still live in GeoMesa - you are using the baseline Z curve implementations as they've been defined and modified here.

If there's a better name for it than something like `hilbert2`, I'm up for a better option. It doesn't make me uncomfortable to reference GeoWave from the codebase, however, especially since the subproject is a straight up pull from geowave-index (in our PR for GeoTrellis, we have also labelled the indexing strategy that leans on this as GeoWave).

/cc @rfecher, not sure if putting GeoWave into the naming makes sense to you, or if you know of a better name that you'd like more.

Reply:
Makes sense to me @lossyrob -- keep the name `geowave` or perhaps `hilbert-geowave`.

Reply:
Going back to this, the underlying Hilbert implementation is the Uzaygezen library.;)